src/audio/host.c

```diff
@@ -170,7 +170,11 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 
 	/* buffer overlap, hardcode host buffer size at the moment */
 	if (hd->local_pos >= hd->host_size)
+#if CONFIG_RENOIR
+		hd->local_pos %= hd->host_size;
+#else
 		hd->local_pos = 0;
+#endif
```

Comment:
we shouldn't have platform-specific ifdefs in common code, that looks likes a missing abstraction.

Reply:
@balakishorepati can you provide more context here. How does the host DMA position work on Renoir ?

Reply:
It is to update the position on wrap of buffer. It is not necessary that on wrap the buffer position would be zero. At wrap, we might have DMAed some partial data from the beginning of the buffer also. In some cases based on the period size, the last frame may not exactly be the end of the buffer and roll over for some bytes in the beginning. So, the modulus operation will give the actual position info. If the the position info is matching with the buffer size, then the modulus operation will give zero, if there are some bytes consumed from the beginning after wrap, then the modulus operation will give the actual position value.
So Instead of make hard coding it to zero, modules operation will give actual position.

Reply:
ok, best to comment this in the source. Fwiw, modulus can be expensive on xtensa, so you may want to work out the position another way if you can.

Reply:
Understand that modulus operation is expensive on Xtensa, but, modulus operation is executed only once at a system memory buffer wrap condition, so believe it won't affect the performance. We will keep an eye on this and look for a better a way.

Reply:
Added generic config to reuse by other platforms.